src/Our.Umbraco.Ditto.Archetype/Models/ArchetypePublishedProperty.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace Our.Umbraco.Ditto.Archetype
+{
+    using global::Archetype.Models;
+    using global::Umbraco.Core.Models;
+
+    public class ArchetypePublishedProperty : IPublishedProperty
+    {
+        private ArchetypePropertyModel _property;
+
+        public ArchetypePublishedProperty(ArchetypePropertyModel property)
+        {
+            _property = property;
+        }
+
+        public object DataValue
+        {
+            get { return this.Value; }
+        }
+
+        public bool HasValue
+        {
+            get
+            {
+                if (_property == null || _property.Value == null)
+                {
+                    return false;
+                }
+
+                return !string.IsNullOrEmpty(_property.Value.ToString());
+            }
+        }
+
+        public string PropertyTypeAlias
+        {
+            get
+            {
+                if (_property != null)
+                {
+                    return _property.PropertyEditorAlias;
+                }
+
+                return null;
+            }
+        }
+
+        public object Value
+        {
+            get
+            {
+                if (this.HasValue)
+                {
+                    return _property.Value;
```

Comment:
I'm struggling with what to return here... whether it should be the raw `_property.Value`, or `_property.GetValue<T>()`?

The problem being that at this point we don't know what the target type should be.

If we go with the raw value, then we skip Archetype's call to `TryConvertWithPropertyValueConverters`...
https://github.com/imulus/Archetype/blob/master/app/Umbraco/Umbraco.Archetype/Models/ArchetypePropertyModel.cs#L39

So far my unit-tests have been with simple `string` values.  I'll aim to try with more complex data models.